mlflow/xgboost/__init__.py

```diff
@@ -668,5 +668,24 @@ def infer_model_signature(input_example):
 
         return model
 
-    safe_patch(FLAVOR_NAME, xgboost, "train", train, manage_run=True)
+    safe_patch(FLAVOR_NAME, xgboost, "train", functools.partial(train, log_models), manage_run=True)
+    safe_patch(
+        FLAVOR_NAME, xgboost.sklearn, "train", functools.partial(train, False), manage_run=True
+    )
```

Comment:
Can we insert a comment explaining why we set `_log_models` to `False` for scikit-learn estimators?

Reply:
Thank you for your suggestion! I will add a comment in the next commit.